src/main/java/net/sf/jabref/shared/DBMSProcessor.java

```diff
@@ -0,0 +1,562 @@
+        this.dbmsHelper = dbmsHelper;
+    }
+
+    /**
+     * Scans the database for required tables.
+     * @return <code>true</code> if the structure matches the requirements, <code>false</code> if not.
+     */
+    public boolean checkBaseIntegrity() {
+        List<String> requiredTables = new ArrayList<>(Arrays.asList("ENTRY", "FIELD", "METADATA")); // the list should be dynamic
+        try {
+            DatabaseMetaData databaseMetaData = dbmsHelper.getMetaData();
+
+            // ...getTables(null, ...): no restrictions
+            try (ResultSet databaseMetaDataResultSet = databaseMetaData.getTables(null, null, null, null)) {
+
+                while (databaseMetaDataResultSet.next()) {
+                    String tableName = databaseMetaDataResultSet.getString("TABLE_NAME").toUpperCase();
+                    requiredTables.remove(tableName); // Remove matching tables to check requiredTables for emptiness
+                }
+
+                return requiredTables.isEmpty();
+            }
+        } catch (SQLException e) {
+            LOGGER.error("SQL Error: ", e);
+        }
+        return false;
+    }
+
+
+    /**
+     * Creates and sets up the needed tables and columns according to the database type.
+     */
+    public void setUpSharedDatabase() {
+        if (dbmsType == DBMSType.MYSQL) {
+            dbmsHelper.executeUpdate(
+                "CREATE TABLE IF NOT EXISTS `ENTRY` (" +
+                "`SHARED_ID` INT(11) NOT NULL PRIMARY KEY AUTO_INCREMENT, " +
+                "`TYPE` VARCHAR(255) NOT NULL, " +
+                "`VERSION` INT(11) DEFAULT 1)");
+
+            dbmsHelper.executeUpdate(
+                "CREATE TABLE IF NOT EXISTS `FIELD` (" +
+                "`ENTRY_SHARED_ID` INT(11) NOT NULL, " +
+                "`NAME` VARCHAR(255) NOT NULL, " +
+                "`VALUE` TEXT DEFAULT NULL, " +
+                "FOREIGN KEY (`ENTRY_SHARED_ID`) REFERENCES `ENTRY`(`SHARED_ID`) ON DELETE CASCADE)");
+
+            dbmsHelper.executeUpdate("CREATE TABLE IF NOT EXISTS `METADATA` (" +
+                    "`KEY` varchar(255) NOT NULL," +
+                    "`VALUE` text NOT NULL)");
+
+        } else if (dbmsType == DBMSType.POSTGRESQL) {
+            dbmsHelper.executeUpdate(
+                "CREATE TABLE IF NOT EXISTS \"ENTRY\" (" +
+                "\"SHARED_ID\" SERIAL PRIMARY KEY, " +
+                "\"TYPE\" VARCHAR, " +
+                "\"VERSION\" INTEGER DEFAULT 1)");
+
+            dbmsHelper.executeUpdate(
+                "CREATE TABLE IF NOT EXISTS \"FIELD\" (" +
+                "\"ENTRY_SHARED_ID\" INTEGER REFERENCES \"ENTRY\"(\"SHARED_ID\") ON DELETE CASCADE, " +
+                "\"NAME\" VARCHAR, " +
+                "\"VALUE\" TEXT)");
+
+            dbmsHelper.executeUpdate("CREATE TABLE IF NOT EXISTS \"METADATA\" ("
+                    + "\"KEY\" VARCHAR,"
+                    + "\"VALUE\" TEXT)");
+
+        } else if (dbmsType == DBMSType.ORACLE) {
```

Comment:
Instead of long if clauses, one can leverage the strategy pattern so that the current instance just knows what to do. In that case, there should be a Oracle, MySQL and Postgres object somehow that just knows how to set up the database, and escape stuff even. So that the dbms specific code is in a dbms specific class, and the other code is in a more general class like this.